typedb/concept/type/thing_type.py

```diff
@@ -18,103 +18,192 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-from typing import Set
 
-from typedb.api.concept.concept import ValueType
-from typedb.api.concept.type.attribute_type import AttributeType
-from typedb.api.concept.type.role_type import RoleType
-from typedb.api.concept.type.thing_type import ThingType, RemoteThingType, Annotation
-from typedb.common.rpc.request_builder import thing_type_set_supertype_req, thing_type_get_instances_req, \
-    thing_type_set_abstract_req, thing_type_unset_abstract_req, thing_type_set_plays_req, thing_type_set_owns_req, \
-    thing_type_get_plays_req, thing_type_get_owns_req, thing_type_unset_plays_req, thing_type_unset_owns_req, \
-    thing_type_get_owns_explicit_req, thing_type_get_owns_overridden_req, thing_type_get_plays_explicit_req, \
-    thing_type_get_plays_overridden, thing_type_get_syntax_req
-from typedb.concept.proto import concept_proto_builder, concept_proto_reader
-from typedb.concept.type.type import _Type, _RemoteType
+from __future__ import annotations
+from abc import abstractmethod, ABC
+from itertools import chain
+from typing import Optional, Iterator, TYPE_CHECKING, Any
+
+from typedb.api.concept.type.thing_type import ThingType
+from typedb.common.exception import TypeDBClientException, UNEXPECTED_NATIVE_VALUE
+from typedb.common.label import Label
+from typedb.common.streamer import Streamer
+from typedb.common.transitivity import Transitivity
+from typedb.concept.thing.thing import _Thing
+from typedb.concept.type.type import _Type
+from typedb.typedb_client_python import Concept as NativeConcept, concept_is_entity_type, concept_is_relation_type, \
+    concept_is_attribute_type, concept_is_root_thing_type, thing_type_is_root, thing_type_is_abstract, \
+    thing_type_get_label, thing_type_delete, thing_type_is_deleted, thing_type_set_label, thing_type_set_abstract, \
+    thing_type_unset_abstract, thing_type_set_plays, thing_type_unset_plays, thing_type_set_owns, thing_type_get_owns, \
+    thing_type_get_plays, thing_type_get_owns_overridden, thing_type_unset_owns, thing_type_get_syntax, \
+    thing_type_get_plays_overridden, concept_iterator_next
+
+if TYPE_CHECKING:
+    from typedb.api.concept.type.attribute_type import AttributeType
+    from typedb.api.concept.value.value import ValueType
+    from typedb.api.concept.type.annotation import Annotation
+    from typedb.concept.type.attribute_type import _AttributeType
+    from typedb.concept.type.role_type import _RoleType
+    from typedb.connection.transaction import _Transaction
+
+
+class _ThingType(ThingType, _Type, ABC):
+
+    @staticmethod
+    def of(concept: NativeConcept):
+        from typedb.concept.type import attribute_type, entity_type, relation_type
+
```

Comment:
Why not import at the top, and why not import the classes? We already need `_AttributeType` for type hints anyway